internal/bminventory/inventory.go

```
@@ -296,24 +297,69 @@ func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params inst
 	}
 
 	return filemiddleware.NewResponder(installer.NewDownloadClusterISOOK().WithPayload(resp.Body),
-		fmt.Sprintf("%s-cluster-%s-discovery.iso", params.ImageID.String(), params.ClusterID.String()))
+		fmt.Sprintf("cluster-%s-discovery.iso", params.ClusterID.String()))
 }
 
-// GenerateClusterISO and return image ID that can be used to download the ISO
 func (b *bareMetalInventory) GenerateClusterISO(ctx context.Context, params installer.GenerateClusterISOParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
 	log.Infof("prepare image for cluster %s", params.ClusterID)
 	var cluster models.Cluster
-	if err := b.db.First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
-		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+
+	tx := b.db.Begin()
+	defer func() {
+		if r := recover(); r != nil {
+			log.Error("generate cluster ISO failed")
+			tx.Rollback()
+		}
+	}()
+
+	if tx.Error != nil {
+		log.WithError(tx.Error).Error("failed to start transaction")
```

Need to return an error, and please fix the `UpdateCluster` place as well.

---

Ah so you found where I copy-pasted from :)

---

i will probably do the same :)